shared-modules-auto-pr/entrypoint.py

```diff
@@ -0,0 +1,308 @@
+    if not shared_modules:
+        log.error(f"No shared-modules found for {app_id}, skipping")
+
+    return shared_modules
+
+
+def parse_github_url(url):
+    """
+    Parses the organization/repo part out of a git remote URL.
+    """
+    if url.startswith("https:"):
+        o = urllib.parse.urlparse(url)
+        return o.path[1:]
+
+    m = _GITHUB_URL_PATTERN.match(url)
+    if m:
+        return m.group("org_repo")
+    else:
+        raise ValueError(f"{url!r} doesn't look like a Git URL")
+
+
+def bump_repository(repo_url: str):
+    log.info(f"Preparing repo {repo_url}")
+    app_id = get_app_id(repo_url)
+    if len(app_id.split(".")) < 1:
+        log.info(f"Repo {repo_url} doesn't seeem to be an app, skipping")
+        return
+
+    repository = clone_repository(repo_url)
+    # If the repository uses a module from shared-modules
+    if prepare_repository(repo_url, repository):
+        log.info(f"Updating repo {repo_url}")
+        update_repository(repository)
+
+
+def open_pr(subject, body, branch):
+    log.info("Opening pull request for branch %s", branch)
+    github_token = os.environ["GITHUB_TOKEN"]
+    g = Github(github_token)
+    user = g.get_user()
+
+    origin_url = (
+        subprocess.check_output(("git", "remote", "get-url", "origin",))
+        .decode("utf-8")
+        .strip()
+    )
+    origin_repo = g.get_repo(parse_github_url(origin_url))
+
+    if origin_repo.permissions.push:
+        log.debug("origin repo is writable")
+        repo = origin_repo
+        fork = None
+    else:
+        log.debug("origin repo not writable; creating fork")
+        fork = user.create_fork(origin_repo)
+        repo = fork
+
+    remote_url = f"https://{github_token}:x-oauth-basic@github.com/{repo.full_name}"
+
+    base = origin_repo.default_branch
+    head = "{}:{}".format(repo.owner.login, branch)
+    pr_message = ((body or "") + "\n\n" + DISCLAIMER).strip()
+    # Include closed PRs – if the maintainer has closed our last PR, we don't want to
+    # open another one.
+    for pr in origin_repo.get_pulls(state="all", base=base, head=head):
+        log.info(
+            "Found existing %s PR: %s",
+            "merged" if pr.is_merged() else pr.state,
+            pr.html_url,
+        )
+        return
+
+    subprocess.check_call(("git", "push", "-u", remote_url, branch))
+
+    pr = origin_repo.create_pull(
+        subject, pr_message, base, head, maintainer_can_modify=True,
+    )
+    log.info("Opened pull request %s", pr.html_url)
+
+
+def update_repository(repo: pygit2.Repository):
+    """
+        Update the repository submodules
+        commit the changes and create a merge request
+    """
+    submodule = repo.lookup_submodule(SHARED_MODULES)
+    shared_modules = submodule.open()
+    message = "Bump shared-modules"
+
+    subprocess.check_call(['git', 'pull', 'origin', 'master'],
+                          cwd=shared_modules.path)
+
+    merge_branch = 'flathubbot/shared-modules'
```

Comment:
You want this to include hash of commit being pushed or else it will cause issues when multiple unmerged updates are stacked up or branch won't be deleted after PR is merged.